android-lib/src/main/java/com/kinvey/android/cache/RealmCache.java

```diff
@@ -501,13 +501,14 @@ void checkAclKmdFields(DynamicRealm realm) {
 
     private String insertOrUpdate(T item, DynamicRealm mRealm){
 
-        T clone = (T)item.clone();
+        T clone = (T)item;
```

Comment:
What is the affect of not cloning the `item`? And why do we have a `clone` variable, if it is just the `item`?